test/e2e/kubectl.go

```diff
@@ -1630,22 +1629,45 @@ var _ = framework.KubeDescribe("Kubectl client", func() {
 })
 
 // Checks whether the output split by line contains the required elements.
-func checkOutput(output string, required [][]string) {
+func checkOutputReturnError(output string, required [][]string) error {
 	outputLines := strings.Split(output, "\n")
 	currentLine := 0
 	for _, requirement := range required {
 		for currentLine < len(outputLines) && !strings.Contains(outputLines[currentLine], requirement[0]) {
 			currentLine++
 		}
 		if currentLine == len(outputLines) {
-			framework.Failf("Failed to find %s in %s", requirement[0], output)
+			return fmt.Errorf("failed to find %s in %s", requirement[0], output)
 		}
 		for _, item := range requirement[1:] {
 			if !strings.Contains(outputLines[currentLine], item) {
-				framework.Failf("Failed to find %s in %s", item, outputLines[currentLine])
+				return fmt.Errorf("failed to find %s in %s", item, outputLines[currentLine])
 			}
 		}
 	}
+	return nil
+}
+
+func checkOutput(output string, required [][]string) {
+	err := checkOutputReturnError(output, required)
+	if err != nil {
+		framework.Failf("%v", err)
+	}
+}
+
+func checkKubectlOutputWithRetry(required [][]string, args ...string) {
+	err := wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+		output := framework.RunKubectlOrDie(args...)
+		err := checkOutputReturnError(output, required)
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
```

Comment:
The only error that comes back from PollImmediate is a timeout error which is far from helpful. I would suggest passing through the original error.
```go
+ var pollErr error
+	err := wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+		output := framework.RunKubectlOrDie(args...)
+		err := checkOutputReturnError(output, required)
+		if err != nil {
+           // log the error
+           pollErr = err
+			return false, nil
+		}
+       pollErr = nil
+		return true, nil
+	})
+ if pollErr != nil {
+       framework.Failf("%v", pollErr)
+ }
```